ckan/authz.py

```diff
@@ -119,6 +121,8 @@ def _get_user(username):
     except TypeError:
         # c is not available
         pass
+    except DetachedInstanceError:
```

Comment:
I can't imagine an instance where catching and passing silently on DetachedInstanceError is the right thing to do.  This is likely to be masking the real error elsewhere. Do you have a deeper stack trace from when this fixed your problem?  I suspect that another query somewhere has left the c.userobj hanging without being connected to a session (somehow). This hasn't shown up in core before, is it possible it's an issue in your extension?

I know we silently pass on the TypeError above, we shouldn't and I'll raise a separate issue for that.